### DISCUSSION TITLE
How to uninstall Kitty and select which terminal emulator is used

### DISCUSSION THREAD

**Discussion by doctorfree** [doctorfree/MusicPlayerPlus#18] [Q&A]
@JOJ0 asks how to uninstall Kitty and wishes to use another terminal emulator.

To uninstall Kitty it is sufficient to simply remove it from `$HOME/.local`

```shell
cd ~/.local
rm -f bin/kitty
rm -f share/applications/kitty-open.desktop
rm -f share/applications/kitty.desktop
rm -f share/man/man1/kitty.1
rm -f share/man/man5/kitty.conf.5
rm -rf kitty.app
```

To select the terminal emulator MusicPlayerPlus will use edit `$HOME/.config/mpprc` and uncomment the preferred MPP_MODE setting. For example, to use the Gnome terminal emulator:

```shell
#MPP_MODE=console
#MPP_MODE=current
MPP_MODE=gnome
#MPP_MODE=retro
#MPP_MODE=simple
#MPP_MODE=tilix
#MPP_MODE=kitty
```

The `MPP_MODE=current` setting indicates to use whatever terminal emulator your are currently running. The recommended terminal emulator and the one I use and test with most frequently is Kitty. The Kitty terminal emulator is the most flexible and supports features not found in other terminal emulators like a remote control socket and scripted terminal control. However, there has been some effort to be terminal agnostic so please open an issue if the terminal emulator you are using does not perform as expected when using MusicPlayerPlus.

**Comment by doctorfree** [comment]
Note that when selecting `MPP_MODE=current` some features are disabled. For example, window and font sizes are not set by MusicPlayerPlus - the current window and font sizes are used. To enable all features one of the supported terminal emulators must be configured. However, current mode should work sufficiently for most.

**Comment by JOJ0** [comment]
In the meantime I think I understood that it does make perfectly sense to install and use a separate terminal emulator to use for musicplayerplus and I'm totaly fine with kitty. Most of my initial problems where not related to kitty but to the missing mpcplus package.

Which terminal emulators are the supported ones?

**Reply by doctorfree** [reply]
Supported terminal emulators are gnome-terminal, tilix, st, cool-retro-term, and kitty. The cool-retro-term is supported mainly as a sort of oddity and not fully functional as it has no way to programmatically control font sizes. 

Kitty is installed by default and used as the fallback if the configured terminal cannot be found. Except on armv7 architectures on which Kitty is not supported - on those systems st (Suckless simple terminal) is installed and used as the fallback.

During `mppinit` a check is performed to see which of the supported terminal emulators are available. If gnome-terminal or tilix are found then a custom terminal profile is installed. This is needed to programmatically control font sizes and other properties. This is probably why your tilix did not have a MusicPlayerPlus profile - it was installed after `mppinit`. Maybe that is a bug.

None of the terminal emulators are installed other than the fallback (Kitty or st). But if they are previously installed and available then they are configured if necessary and made available. Supporting all these terminal emulators turned out to be a real pain. I wish I had just forced Kitty and ignored the rest.

**Reply by doctorfree** [reply]
Note that you can create MusicPlayerPlus terminal profiles for terminal emulators you installed after running `mppinit` by re-running mppinit with the profiles argument: `mppinit profiles`

**Reply by doctorfree** [reply]
Oops, one other gotcha. The `dconf` package is required in order to create terminal profiles. On Arch and Arch-like distributions `dconf` is an extra package and not necessarily installed. It may be necessary to install `dconf` on Arch-like distributions before running `mppinit` or `mppinit profiles`. I will look into how to automate this.

On non-Arch platforms or for those Arch users who configure Kitty as the terminal emulator  this is not an issue.
